components/proxy/src/main/java/com/hotels/styx/startup/StyxPipelineFactory.java

```diff
@@ -66,18 +67,21 @@ public HttpHandler create(StyxServerComponents config) {
 
         boolean requestTracking = config.environment().configuration().get("requestTracking", Boolean.class).orElse(false);
 
+        List<NamedPlugin> plugins = config.environment().configStore()
+                .valuesStartingWith("plugins", NamedPlugin.class);
```

Comment:
Consider config-store as global data,  and therefore avoid unnecessary lookups when possible. 
In this case I would pass `plugins` list as a method argument. It is already available at the call site.